### PR TITLE
fix: create spec for cra

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes https://cypress-io.atlassian.net/browse/UNIFY-1179

### User facing changelog

Use correct file glob for Create React App.

### Additional details

CRA uses `.js` for components, so we need to include that in the default glob we use to file components.

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue (or this PR, if no issue exists) been tagged with a release in ZenHub? (user-facing changes only)
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

File: packages/app/cypress/e2e/create-spec.cy.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import type { e2eProjectDirs } from '@packages/frontend-shared/cypress/e2e/support/e2eProjectDirs'
+
+function scaffoldAndVisit(project: typeof e2eProjectDirs[number], type: 'component' | 'e2e') {
```
Comment (lmiller1990) [inline packages/app/cypress/e2e/create-spec.cy.ts:3]: we need these tests for every combo of project, most likely, like nextjs, etc etc.

File: system-tests/projects/create-react-app-js/src/Foo.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import React from 'react'
+
+export default function Foo () {
```
Comment (lmiller1990) [inline system-tests/projects/create-react-app-js/src/Foo.js:3]: We don't support codegen for non default exports yet, but I'd like to see support for this soon.

File: packages/app/cypress/e2e/create-spec.cy.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  cy.scaffoldProject(project)
+  cy.openProject(project)
+  cy.startAppServer(type)
+  cy.__incorrectlyVisitAppWithIntercept()
```
Comment (ZachJW34) [inline packages/app/cypress/e2e/create-spec.cy.ts:7]: Any reason to use `__incorrectlyVisitAppWithIntercept` vs `visitApp`?

**Review by ZachJW34 — APPROVED** [review]
Works great!

**Review by elylucas — APPROVED** [review]

File: packages/app/cypress/e2e/create-spec.cy.ts (hunk 3 above)
Comment (lmiller1990) [reply packages/app/cypress/e2e/create-spec.cy.ts:7]: I have no idea what this does differently - I just copied pasted this from the other tests that do the same thing. I'd like to find at some point, but for now it seems like this is how tests in `app` look. Maybe we should create a custom command at some point.

**Comment by lmiller1990** [comment]
fixed here https://github.com/cypress-io/cypress/pull/20306

**Pull request closed without merging**